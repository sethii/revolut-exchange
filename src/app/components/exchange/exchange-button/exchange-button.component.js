import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { setPropTypes } from 'recompose';

const Button = styled.button`
  border: none;
  border-radius: 25px;
  color: white;
  background-color: rgba(247, 0, 135, 1);
  width: 600px;
  font-size: 18px;
  padding: 20px 10px;
  
  &:hover {
    cursor: pointer;
  }  
  
  &:disabled {
    background-color: rgba(247, 0, 135, 0.28);
  }
`;

const enhance = setPropTypes({
  disabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired
});

export const ExchangeButton = enhance(({ onClick, disabled }) => <Button onClick={onClick} disabled={disabled}>
  Exchange
</Button>);
