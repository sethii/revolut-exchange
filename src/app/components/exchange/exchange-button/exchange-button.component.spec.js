import React from 'react';
import renderer from 'react-test-renderer';
import { toClass } from 'recompose';
import { mount } from 'enzyme';

import { ExchangeButton } from './exchange-button.component';

describe('Exchange button component', () => {
  it('renders exchange button', () => {
    const ExchangeButtonComponent = toClass(ExchangeButton);

    const onClick = jest.fn();

    expect(renderer.create(<ExchangeButtonComponent disabled={true} onClick={onClick}/>).toJSON()).toMatchSnapshot();
  });

  it('handles clicks if not disabled', () => {
    const ExchangeButtonComponent = toClass(ExchangeButton);
    const onClick = jest.fn();
    const wrapper = mount(<ExchangeButtonComponent disabled={false} onClick={onClick}/>);

    wrapper.find('button').simulate('click');

    expect(onClick).toBeCalled();
  });

  it('handles clicks if disabled', () => {
    const ExchangeButtonComponent = toClass(ExchangeButton);
    const onClick = jest.fn();
    const wrapper = mount(<ExchangeButtonComponent disabled={true} onClick={onClick}/>);

    wrapper.find('button').simulate('click');

    expect(onClick).not.toBeCalled();
  });
});
