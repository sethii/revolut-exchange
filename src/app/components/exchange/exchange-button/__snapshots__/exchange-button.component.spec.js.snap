// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange button component renders exchange button 1`] = `
.glamor-1 {
  border: none;
  border-radius: 25px;
  color: white;
  background-color: rgba(247, 0, 135, 1);
  width: 600px;
  font-size: 18px;
  padding: 20px 10px;
}

.glamor-1:hover {
  cursor: pointer;
}

.glamor-1:disabled {
  background-color: rgba(247, 0, 135, 0.28);
}

<button
  className="glamor-0 glamor-1"
  disabled={true}
  onClick={[Function]}
>
  Exchange
</button>
`;
